### PR TITLE
Read vectored metric

### PR DESCRIPTION
**Pull request by singhravidutt**

### CODE REVIEW

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java [hunk 1]
```diff
@@ -222,7 +224,6 @@ public synchronized int read(@Nonnull byte[] buf, int offset, int length) throws
             if (numRead > 0) {
               // -1 means we actually read 0 bytes, but requested at least one byte.
               totalBytesRead += numRead;
-              statistics.incrementBytesRead(numRead);
               statistics.incrementReadOps(1);
```
Comment (arunkumarchacko) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java:227]: will this get updated correctly?

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/VectoredIOImpl.java [hunk 2]
```diff
@@ -101,202 +97,259 @@ public void readVectored(
...
-      IntFunction<ByteBuffer> allocate,
-      ReadChannelProvider channelProvider)
-      throws IOException {
-    List<? extends FileRange> sortedRanges = validateNonOverlappingAndReturnSortedRanges(ranges);
-    for (FileRange range : ranges) {
-      // TODO(user): upgrade to use validateAndSortRanges
-      validateRangeRequest(range);
-      CompletableFuture<ByteBuffer> result = new CompletableFuture<>();
-      range.setData(result);
+  class VectoredReadChannel {
```
Comment (arunkumarchacko) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/VectoredIOImpl.java:108]: static?

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/VectoredIOImpl.java [hunk 3]
```diff
@@ -101,202 +97,259 @@ public void readVectored(
...
-    try (SeekableByteChannel channel = channelProvider.getReadChannel()) {
-      channel.position(range.getOffset());
-      ByteBuffer dst = allocate.apply(range.getLength());
-      int numRead = channel.read(dst.duplicate());
-      if (numRead < range.getLength()) {
-        throw new EOFException(
-            String.format(
-                "EOF reached before whole range can be read, range: %s, path: %s",
-                range, channelProvider.gcsPath));
+    private class ReadChannelProvider {
```
Comment (arunkumarchacko) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/VectoredIOImpl.java:313]: static

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java (hunk 1 above)
Comment (singhravidutt) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java:227]: It is getting validated by some unit test. Added a separate integration test to verify readBytes metric.

**Review by arunkumarchacko — APPROVED** [review]

**Pull request merged**